examples/codenames/Codenames_RL.ipynb

```diff
@@ -0,0 +1,12542 @@
+          "output_type": "stream",
+          "text": [
+            "INFO 05-06 06:35:22 model_runner.py:1115] Loading model weights took 2.2265 GB\n",
+            "INFO 05-06 06:35:22 punica_selector.py:18] Using PunicaWrapperGPU.\n",
+            "INFO 05-06 06:35:27 worker.py:267] Memory profiling takes 4.73 seconds\n",
+            "INFO 05-06 06:35:27 worker.py:267] the current vLLM instance can use total_gpu_memory (79.20GiB) x gpu_memory_utilization (0.78) = 62.10GiB\n",
+            "INFO 05-06 06:35:27 worker.py:267] model weights take 2.23GiB; non_torch_memory takes 0.16GiB; PyTorch activation peak memory takes 2.71GiB; the rest of the memory reserved for KV Cache is 57.00GiB.\n",
+            "INFO 05-06 06:35:27 executor_base.py:111] # cuda blocks: 103764, # CPU blocks: 10922\n",
+            "INFO 05-06 06:35:27 executor_base.py:116] Maximum concurrency for 32768 tokens per request: 50.67x\n",
+            "INFO 05-06 06:35:31 model_runner.py:1434] Capturing cudagraphs for decoding. This may lead to unexpected consequences if the model is not static. To run the model in eager mode, set 'enforce_eager=True' or use '--enforce-eager' in the CLI. If out-of-memory error occurs during cudagraph capture, consider decreasing `gpu_memory_utilization` or switching to eager mode. You can also reduce the `max_num_seqs` as needed to decrease memory usage.\n"
+          ]
+        },
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "Capturing CUDA graph shapes: 100%|██████████| 49/49 [00:26<00:00,  1.84it/s]\n"
+          ]
+        },
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "INFO 05-06 06:35:58 model_runner.py:1562] Graph capturing finished in 27 secs, took 1.30 GiB\n",
+            "INFO 05-06 06:35:58 llm_engine.py:436] init engine (profile, create kv cache, warmup model) took 35.75 seconds\n"
+          ]
+        },
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "Unsloth 2025.3.19 patched 36 layers with 36 QKV layers, 36 O layers and 36 MLP layers.\n"
+          ]
+        }
+      ],
+      "source": [
+        "# ───────────────────────────  7. Creating the model  ───────────────────────────\n",
+        "from art.local import LocalBackend\n",
+        "from openai import AsyncOpenAI\n",
+        "\n",
+        "model = art.TrainableModel(\n",
+        "    name=MODEL_NAME,\n",
+        "    project=\"codenames-rl\",\n",
+        "    base_model=BASE_MODEL_NAME,\n",
+        ")\n",
+        "backend = LocalBackend(path=\"./.art\")\n",
+        "await model.register(backend)\n",
+        "\n",
+        "art_client = model.openai_client()\n",
+        "openai_client = AsyncOpenAI()\n",
+        "model_dict = {\"client\": art_client, \"name\": MODEL_NAME}\n",
+        "benchmark_model_dict = {\"client\": openai_client, \"name\": VAL_BENCHMARK_MODEL}"
```

Comment:
Instead of creating this model_dict, I think it would be cleaner to declare an art.TrainableModel and art.Model instances.